system/scripts/zone/ais/basic_monster.cs

```diff
@@ -89,7 +90,7 @@ protected IEnumerable StopAndAttack()
 	private void CheckEnemies()
 	{
 		var mostHated = GetMostHated();
-		if (mostHated != null)
+		if (mostHated != null && (target == null || mostHated != target))
```

Comment:
Quick question, wouldn't the check in `CheckTarget` be enough? We're calling that function anyway, even the name matches, and we could just do this, no?
```text
if target != GetMostHated
    changeTarget
```

Reply:
I changed the code a little to make it more straight forward, with a single check in `CheckTarget`, where we're already handling the current target and simply change it if applicable. Let me know if I missed something in your implementation.